src/lib/instruction-parser.ts

```diff
@@ -258,19 +293,62 @@ export class InstructionParser {
               outAccount,
             });
           }
-        });
+        }
         break;
       }
     }
-
     return { swaps, innerInstructions };
   }
 
+  // @note swap direction must be determined here if any new custom swaps require it
+  getCustomSwap(
+    innerInstructions: (PartialInstruction | ParsedInstruction)[],
+    index: number,
+    swapProgramId: PublicKey,
+    swapIxName: string,
+    swapData: Swap
+  ) {
+    const positions = SWAP_IN_OUT_ACCOUNTS_POSITION[swapIxName];
+    for (let i = index + 1; i < innerInstructions.length; i++) {
+      if (
+        innerInstructions[i].programId.toBase58() == swapProgramId.toBase58()
+      ) {
+        let inAccount: PublicKey;
+        let outAccount: PublicKey;
+        if (swapIxName == 'openbook' || swapIxName == 'serum') {
+          const newOrderInstruction = innerInstructions[index];
+          const settleFundsInstruction = innerInstructions[i]
+          const sideObj = Object.values(swapData)[0]["side"];
+          const side = Object.keys(sideObj)[0];
+          inAccount = (newOrderInstruction as any).accounts[positions.in];
+          outAccount = (settleFundsInstruction as any).accounts[positions.out[side]];
+          return {
+            inAccount,
+            outAccount,
+            skipToIndex: i,
+          };
+        }
```

Comment:
`} else if {`

Reply:
Openbook and Serum similar instructions for swap. So for both cases, swap can be extracted in the same way
https://github.com/jup-ag/jupiter-aggregator-program/blob/a13ddd4148be909269b46e66ce148156b7d39d3b/programs/jupiter/src/process_swap.rs#L346